### PR TITLE
Measure shutdown time

### PR DESCRIPTION
**Pull request by leivos-unikie**
Add test case for measuring shutdown time. Modify `Verify shutdown via serial` to prefer shutdown detection by serial for all targets but still allow fallback to detection via network. Skip shutdown time measurement because of inaccuracy if shutdown detection via serial was not possible.

Measure shutdown time also by power if power measurement setup is available (prod lenovo-x1 only). If shutdown time measured by serial and power differ more than 10 sec fail the test. Shutdown measurement by power can be considered as the most accurate method of measuring time to HW power off. It can be useful in detecting edge cases on lenovo-x1 where buggy shutdown renders the HW partially powered although all communication with the device is lost. In normal conditions we expect 'Measure Shutdown Time' and 'Measure Shutdown Time By Power' results not to be very far from each other (in my tests there was < 5sec difference).

orin-agx-64
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/6819/
orin-agx-64 regression
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/6849/

orin-nx
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/6835/

lenovo-x1
https://ci-prod.vedenemo.dev/job/ghaf-hw-test-manual/2211/
https://ci-prod.vedenemo.dev/job/ghaf-hw-test-manual/2212/

Regression check on darter-pro
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/6821/

### CODE REVIEW

File: Robot-Framework/test-suites/performance-tests/boot_time.robot [hunk 1]
```diff
@@ -33,6 +35,18 @@ Measure Soft Boot Time
     Soft Reboot Device
     Get Boot times
 
+Measure Shutdown Time
+    [Documentation]  Measure how long it takes to device to shut down with software shutdown
+    [Tags]           SP-T83  SP-T83-1  lenovo-x1  dell-7330  lab-only
```
Comment (milva-unikie) [inline Robot-Framework/test-suites/performance-tests/boot_time.robot]: Why is darter-pro not included?

File: Robot-Framework/test-suites/performance-tests/boot_time.robot [hunk 2]
```diff
@@ -145,14 +211,47 @@ Log Journal To Debug
...
+        EXCEPT
+            Log    Ignoring invalid measured power sample    console=True
+        END
+        Sleep  0.5
+    END
+
 Boot Time Test Teardown
     Run Keyword If Test Failed   Failed Boot Time Test Teardown
     IF   ${IS_LAPTOP}    Login to laptop
+    Prepare Test Environment
```
Comment (milva-unikie) [inline Robot-Framework/test-suites/performance-tests/boot_time.robot]: Why `Prepare Test Environment`? It is only meant to be used in suite setups. Using it here will mess up the suite logging.

File: Robot-Framework/test-suites/performance-tests/boot_time.robot [hunk 3]
```diff
@@ -145,14 +211,47 @@ Log Journal To Debug
...
 
 Failed Boot Time Test Teardown
     Hard Reboot Device And Connect
     IF   ${IS_LAPTOP}
         Switch to vm          ${HOST}
         Log Journal To Debug  boot=-1
     END
+    Prepare Test Environment
```
Comment (milva-unikie) [inline Robot-Framework/test-suites/performance-tests/boot_time.robot]: Same question here

File: Robot-Framework/test-suites/performance-tests/boot_time.robot [hunk 4]
```diff
@@ -145,14 +211,47 @@ Log Journal To Debug
...
 
+Shutdown Time Teardown
+    Close All Connections
+    Delete All Ports
+    Set Global Variable    ${UART_CAPTURE_ACTIVE}    ${False}
+    Sleep  10
+    Hard Reboot Device And Connect
+    Prepare Test Environment
```
Comment (milva-unikie) [inline Robot-Framework/test-suites/performance-tests/boot_time.robot]: Same question here

File: Robot-Framework/test-suites/performance-tests/boot_time.robot [hunk 5]
```diff
@@ -145,14 +211,47 @@ Log Journal To Debug
...
         Log Journal To Debug  boot=-1
     END
+    Prepare Test Environment
 
+Shutdown Time Teardown
+    Close All Connections
+    Delete All Ports
+    Set Global Variable    ${UART_CAPTURE_ACTIVE}    ${False}
+    Sleep  10
+    Hard Reboot Device And Connect
```
Comment (milva-unikie) [inline Robot-Framework/test-suites/performance-tests/boot_time.robot]: This could cause issues because device is already turned off. Sometimes the OFF-press will also turn the laptop on and then ON-press will turn it off. To solve this in boot test there is second ON if first did not work. I think we need something similar here.

    IF    ${IS_AVAILABLE} == False
        Log To Console    Turning device on again...
        Turn Laptop On
        Check If Device Is Up    retry=110s
    END

**Comment by leivos-unikie** [comment]
Tested again after last changes:

Lenovo-X1 (with power measurement)
https://ci-prod.vedenemo.dev/job/ghaf-hw-test-manual/2416/
image

Darter-Pro
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/7358/
image

Orin-AGX-64
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/7356/

Orin-NX
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/7357/

File: Robot-Framework/resources/device_control.resource [hunk 6]
```diff
@@ -156,14 +165,14 @@ Wake up device
...
-    IF  not ${IS_LAPTOP} or "${SERIAL_PORT}" == "NONE"
+    IF  "${SERIAL_PORT}" == "NONE"
         IF  ${power_off}
             ${time}   Verify shutdown via network    required_down_count=1
         ELSE
             ${time}   Verify shutdown via network
         END
     ELSE
-        ${time}   Verify shutdown via serial
+        ${time}  ${verified_via_network}   Verify shutdown via serial
```
Comment (azbeleva) [inline Robot-Framework/resources/device_control.resource]: why returned variable  ${verified_via_network}?

**Comment by leivos-unikie** [comment]
New tests run

Regression on darter-pro
https://ci-dev.vedenemo.dev/job/ghaf-hw-test-manual/7385/
(few failed tests but nothing related to this)

Shutdown time test on lenovo-x1
https://ci-prod.vedenemo.dev/job/ghaf-hw-test-manual/2417/
image

File: Robot-Framework/test-suites/performance-tests/boot_time.robot (hunk 5 above)
Comment (leivos-unikie) [reply Robot-Framework/test-suites/performance-tests/boot_time.robot]: Improved.

File: Robot-Framework/test-suites/performance-tests/boot_time.robot (hunk 2 above)
Comment (leivos-unikie) [reply Robot-Framework/test-suites/performance-tests/boot_time.robot]: Not using that anymore.

File: Robot-Framework/test-suites/performance-tests/boot_time.robot (hunk 3 above)
Comment (leivos-unikie) [reply Robot-Framework/test-suites/performance-tests/boot_time.robot]: Not using that anymore.

File: Robot-Framework/test-suites/performance-tests/boot_time.robot (hunk 4 above)
Comment (leivos-unikie) [reply Robot-Framework/test-suites/performance-tests/boot_time.robot]: Not using that anymore.

File: Robot-Framework/test-suites/performance-tests/boot_time.robot (hunk 1 above)
Comment (leivos-unikie) [reply Robot-Framework/test-suites/performance-tests/boot_time.robot]: Also Darter included now.

File: Robot-Framework/resources/device_control.resource (hunk 6 above)
Comment (leivos-unikie) [reply Robot-Framework/resources/device_control.resource]: fixed
